examples/stock/company.v.py

```diff
@@ -21,101 +22,106 @@ def __init__(_company: address, _total_shares: currency_value,
 
     self.company = _company
     self.total_shares = _total_shares
-
     self.price = initial_price
 
-    # Company holds all the shares at first, but can sell them all
+    # The company holds all the shares at first, but can sell them all.
     self.holdings[self.company] = _total_shares
 
 @public
 @constant
 def stock_available() -> currency_value:
     return self.holdings[self.company]
 
-# Give value to company and get stock in return
+# Give some value to the company and get stock in return.
 @public
 @payable
 def buy_stock():
     # Note: full amount is given to company (no fractional shares),
-    #       so be sure to send exact amount to buy shares
+    #       so be sure to send the exact amount to buy shares.
+    # Note that this could be implemented as a check for the buyer, 
+    #       that throws if the amount is not equal to a whole 
+    #       number of shares.
     buy_order = msg.value / self.price # rounds down
 
-    # There are enough shares to buy
+    # Check that there are enough shares to buy.
     assert self.stock_available() >= buy_order
 
-    # Take the shares off the market and give to stockholder
+    # Take the shares off the market and give them to the stockholder.
     self.holdings[self.company] -= buy_order
     self.holdings[msg.sender] += buy_order
 
-    # Log the buy event
+    # Log the buy event.
     log.Buy(msg.sender, buy_order)
 
-# So someone can find out how much they have
+# Find out how much stock any address (that's owned by someone) has.
 @public
 @constant
 def get_holding(_stockholder: address) -> currency_value:
     return self.holdings[_stockholder]
 
-# The amount the company has on hand in cash
+# Return the amount the company has on hand in cash.
 @public
 @constant
 def cash() -> wei_value:
     return self.balance
 
-# Give stock back to company and get my money back!
+# Give stock back to the company and get money back as ETH.
 @public
 def sell_stock(sell_order: currency_value):
-    assert sell_order > 0 # Otherwise, will fail at send() below
-    # Can only sell as much stock as you own
+    assert sell_order > 0 # Otherwise, this would fail at send() below, 
+        # due to an OOG error (there would be zero value available for gas).
```

Comment:
We're using the `REVERT` opcode for `assert` and `send` so gas should be returned when it throws.

Reply:
OK, I'll edit the comment.